lib/circuitry/provisioning.rb

```diff
@@ -0,0 +1,59 @@
+require 'circuitry/provisioning/queue_creator'
+require 'circuitry/provisioning/topic_creator'
+require 'circuitry/provisioning/subscription_creator'
+
+module Circuitry
+  module Provisioning
+    class << self
+      def provision_from_config(config, logger: Logger.new(STDOUT))
```

Comment:
It's also a bit tedious that you're having to pass `logger` to every other method in this file.  Why not keep this as an instantiable class instead of a module so that you can assign config and logger to attr_accessors?